src/backend/marsha/core/management/commands/clean_mediapackages.py

```diff
@@ -0,0 +1,134 @@
+"""Clean orphan mediapackages management command."""
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+import boto3
+
+
+aws_credentials = {
+    "aws_access_key_id": settings.AWS_ACCESS_KEY_ID,
+    "aws_secret_access_key": settings.AWS_SECRET_ACCESS_KEY,
+    "region_name": settings.AWS_S3_REGION_NAME,
+}
+
+# Configure medialive client
+medialive_client = boto3.client("medialive", **aws_credentials)
+
+# Configure mediapackage client
+mediapackage_client = boto3.client("mediapackage", **aws_credentials)
+
+
+def _recursive_call(  # nosec
+    function, key, params=None, next_token=None, next_token_key="NextToken"
+):
+    """
+    Recursive generic call.
+
+    Returns a list of items from the same key in response.
+    """
+    if not params:
+        params = {}
+
+    if next_token:
+        params[next_token_key] = next_token
+
+    response = function(**params)
+    items = response.get(key)
+    next_token = response.get(next_token_key)
+
+    if next_token:
+        items.extend(_recursive_call(function, key, params, next_token, next_token_key))
+    return items
+
+
+def list_mediapackage_channels():
+    """List all mediapackage channels."""
+    return _recursive_call(mediapackage_client.list_channels, key="Channels")
+
+
+def list_mediapackage_channel_harvest_jobs(channel_id):
+    """List all harvest jobs for a mediapackage channel."""
+    return _recursive_call(
+        mediapackage_client.list_harvest_jobs,
+        key="HarvestJobs",
+        params={"IncludeChannelId": channel_id},
+    )
+
+
+def list_mediapackage_channel_origin_endpoints(channel_id):
+    """List all origin endpoints for a mediapackage channel."""
+    return _recursive_call(
+        mediapackage_client.list_origin_endpoints,
+        key="OriginEndpoints",
+        params={"ChannelId": channel_id},
+    )
+
+
+def list_medialive_channels():
+    """List all medialive channels."""
+    return _recursive_call(medialive_client.list_channels, key="Channels")
+
+
+def list_indexed_medialive_channels():
+    """List and index all medialive channels by their name."""
+    return {
+        medialive_channel.get("Name"): medialive_channel
+        for medialive_channel in list_medialive_channels()
+    }
+
+
+def delete_mediapackage_channel(channel_id):
+    """Delete a mediapackage channel and related enpoints."""
+    origin_endpoints = list_mediapackage_channel_origin_endpoints(channel_id=channel_id)
+    deleted_endpoints = []
+    for origin_endpoint in origin_endpoints:
+        mediapackage_client.delete_origin_endpoint(Id=origin_endpoint.get("Id"))
+        deleted_endpoints.append(origin_endpoint.get("Id"))
+    mediapackage_client.delete_channel(Id=channel_id)
+    return deleted_endpoints, channel_id
```

Comment:
I don't understand why you would like to return `channel_id` ? It's an input parameter, the calling function already knows it.